rfcs/0000-execute-builds-in-standard-containers.md

```diff
@@ -0,0 +1,132 @@
+be routed through the proxy. Due to the fact that init containers run before any normal container can start, kpack
+builds are unable to reach the network because they are attempting to communicate through the envoy proxy container that
+has not and will not start until the build completes.
+
+To get around this issue, kpack currently applies an annotation (`sidecar.istio.io/inject=false`) to all build pods to
+disable Istio sidecar injection. This presents a problem where users might want to take advantage of the various
+features of Istio, but cannot because kpack is bypassing it.
+
+[buildpack]: https://buildpacks.io/
+
+[cnb-lifecycle]: https://buildpacks.io/docs/concepts/components/lifecycle/
+
+[envoy-proxy]: https://www.envoyproxy.io/
+
+[init-containers]: https://kubernetes.io/docs/concepts/workloads/pods/init-containers/
+
+[istio]: https://istio.io/
+
+[istio-sidecar-injection]: https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/#injection
+
+[istio-service-mesh]: https://istio.io/latest/about/service-mesh/#what-is-a-service-mesh
+
+[istio-init-container-compatability]: https://istio.io/latest/docs/setup/additional-setup/cni/#compatibility-with-application-init-containers
+
+## Outcome
+
+Kpack builds can execute successfully when an Istio sidecar when injected into a build pod.
+
+## Actions to take
+
+1. Create a new binary called `build-waiter`. This will have 2 primary responsibilities/modes of operation:
+    1. Copy Mode - This mode will be executed in an init container to copy the binary into a mounted volume that will
+       be mounted to every other container in the pod
+    2. Wait Mode - This mode will wait for a particular file to exist (For this implementation it will be a file
+       named for the step we are waiting on inside the shared mount that the `build-waiter` binary is in) and once it
+       does, it will execute the provided command. Upon completion of the command execution it will write out a file
+       corresponding to the build step that was just completed. If the command fails, the error will be written to an
+       error file that is being watched by every other container. If the error file is created, all other containers
+       will exit with the same error code, forcing the pod to enter a failed state. There is also a configurable
+       timeout period that will specify how long the container is going to wait for th eprevious steps to complete
+       before exiting with an error. Note: For `prepare` we can provide an empty `wait step` so that it begins
+       executing immediately.
+
+   Example of possible `build-waiter` implementation is in the [spike][kpack-spike-build-waiter].
+
+2. Convert the existing init containers in `build_pod.go` to be regular containers and add an init container to copy
+   the `build-waiter` to a shared mount
+3. Using a similar [step modifier][step-modifier] approach , we can put this new behavior behind a feature gate and
```

Comment:
This is getting extremely implementation specific which I believe is not important for the underlying RFC. Would it be possible to call out just the expected outcome?